x/globalfee/types/code_authorization_test.go

```diff
@@ -0,0 +1,61 @@
+package types_test
+
+import (
+	"testing"
+
+	"github.com/public-awesome/stargaze/v9/x/globalfee/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCodeAuthorizationValidate(t *testing.T) {
+	testCases := []struct {
+		testCase    string
+		ca          types.CodeAuthorization
+		expectError bool
+	}{
+		{
+			"fail: empty methods",
+			types.CodeAuthorization{
+				CodeId:  1,
+				Methods: []string{},
+			},
+			true,
+		},
+		{
+			"fail: invalid method names",
+			types.CodeAuthorization{
+				CodeId:  1,
+				Methods: []string{"^&()"},
+			},
+			true,
+		},
+		{
+			"ok: valid name",
+			types.CodeAuthorization{
+				CodeId:  1,
+				Methods: []string{"mint"},
+			},
+			false,
+		},
+		{
+			"ok: wildcard",
+			types.CodeAuthorization{
+				CodeId:  1,
+				Methods: []string{"*"},
+			},
+			false,
+		},
```

Comment:
add snake_case and with spaces  tests cases.

This might be a good candidate for fuzzing. Although not required